### PR TITLE
gee: Add call to meson.override_dependency

### PR DESCRIPTION
**Pull request by hrxi**

### CODE REVIEW

File: subprojects/packagefiles/gee/gee/meson.build [hunk 1]
```diff
@@ -106,3 +106,4 @@ gee_dep = declare_dependency(
     dependencies: gee_deps,
     include_directories: include_directories('.')
 )
+meson.override_dependency('gee-0.8', gee_dep)
```
Comment (jpakkane) [inline subprojects/packagefiles/gee/gee/meson.build:109]: This is not strictly speaking needed. The dependency lookup uses the value in the .wrap file. `override_dependency` is only used in special cases (e.g. when package `foo` provides an unrelated dependency `bar` and you want that to be used as opposed to the original library that provides `bar`).

Comment (xclaesse) [reply subprojects/packagefiles/gee/gee/meson.build:109]: > Not sure how that passed the CI, I thought we were checking for that.

Ah! missing from releases.json, so it did not get tested at all! Which explains why locally this wrap fails immediately with `ERROR: Target 'gee-0.8' has 1 outputs: ['libgee-0.8.a'], but only 4 "install_dir"s were found.`.

File: releases.json [hunk 2]
```diff
@@ -705,6 +705,14 @@
       "1.14.1-1"
     ]
   },
+  "gee": {
+    "dependency_names": [
+      "gee-0.8"
+    ],
+    "versions": [
+      "0.20.6"
```
[the remark this answers is not in the document]
Comment (hrxi) [reply releases.json]: Thanks!

**Comment by xclaesse** [comment]
@hrxi you'll have to add `valac-bin` debian package in ci_config.json. Maybe skip on other platforms, unless valac can be installed with brew and choco?

File: tools/sanity_checks.py [hunk 3]
```diff
@@ -383,6 +383,9 @@ def check_new_release(self, name: str, builddir: str = '_build', deps=None, prog
                     if 'not found' in error:
                         print('cannot verify in wrapdb due to missing dependency')
                         return
+                elif 'ERROR: Could not execute Vala compiler: valac' in error:
+                    print('cannot verify in wrapdb due to missing dependency')
+                    return
```
Comment (xclaesse) [inline tools/sanity_checks.py:388]: TBH, I'm frustrated that it still try to build on platforms we explicitly say won't work. @eli-schwartz forced adding that but I disagree, we just end up adding all kind of reasons here and waste time on it every time a new reason appears. Anyway... I guess that's fine...

**Comment by hrxi** [comment]
@xclaesse I'm at loss here.

```
"cc" "-Isubprojects/libgee-0.20.6/gee/libgee-0.8-2.6.1.dll.p" "-Isubprojects/libgee-0.20.6/gee" "-I../subprojects/libgee-0.20.6/gee" "-Isubprojects/libgee-0.20.6/utils" "-I../subprojects/libgee-0.20.6/utils" "-ID:/a/_temp/msys64/clang32/include/glib-2.0" "-ID:/a/_temp/msys64/clang32/lib/glib-2.0/include" "-fcolor-diagnostics" "-D_FILE_OFFSET_BITS=64" "-w" "-O0" "-g" -MD -MQ subprojects/libgee-0.20.6/gee/libgee-0.8-2.6.1.dll.p/meson-generated_abstractmultimap.c.obj -MF "subprojects/libgee-0.20.6/gee/libgee-0.8-2.6.1.dll.p/meson-generated_abstractmultimap.c.obj.d" -o subprojects/libgee-0.20.6/gee/libgee-0.8-2.6.1.dll.p/meson-generated_abstractmultimap.c.obj "-c" subprojects/libgee-0.20.6/gee/libgee-0.8-2.6.1.dll.p/abstractmultimap.c
../subprojects/libgee-0.20.6/gee/abstractmultimap.vala:243:19: error: incompatible function pointer types assigning to 'gboolean (*)(GeeIterator *)' (aka 'int (*)(struct _GeeIterator *)') from 'gboolean (*)(GeeAbstractMultiMapMappingIterator *)' (aka 'int (*)(struct _GeeAbstractMultiMapMappingIterator *)') [-Wincompatible-function-pointer-types]
        iface->get_valid = (gboolean (*) (GeeAbstractMultiMapMappingIterator*)) gee_abstract_multi_map_mapping_iterator_get_valid;
                         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The compiler is called with `-w` which should squelch all warnings, yet it outputs a warning **as error**. I'm not sure how to fix this. In my local test setup, clang respects the `-w` flag on MSYS2, and on Linux.

Do you have an idea on how to fix this?

**Comment by neheb** [comment]
Needs rebasing?

**Comment by hrxi** [comment]
> Needs rebasing?

Needs a fix for the problem in the comment: https://github.com/mesonbuild/wrapdb/pull/1098#issuecomment-1673496776.

**Comment by neheb** [comment]
try setting warning_level to 0.

**Comment by hrxi** [comment]
Yay. All green now, thanks to Mangix from IRC and an unnamed friend of mine. :)

**Pull request merged**